authentication/src/main/java/io/camunda/authentication/session/WebSessionMapper.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.session;
+
+import io.camunda.search.entities.PersistentWebSessionEntity;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.support.GenericConversionService;
+import org.springframework.core.serializer.support.DeserializingConverter;
+import org.springframework.core.serializer.support.SerializingConverter;
+
+public class WebSessionMapper {
+
+  private final WebSessionAttributeConverter converter;
+
+  public WebSessionMapper(final WebSessionAttributeConverter converter) {
+    this.converter = converter;
+  }
+
+  public PersistentWebSessionEntity toPersistentWebSession(final WebSession session) {
+    final var attributes = serializeSessionAttributes(session);
+    return new PersistentWebSessionEntity(
+        session.getId(),
+        session.getCreationTime().toEpochMilli(),
+        session.getLastAccessedTime().toEpochMilli(),
+        session.getMaxInactiveInterval().getSeconds(),
+        attributes);
+  }
+
+  public WebSession fromPersistentWebSession(final PersistentWebSessionEntity entity) {
+    final var sessionId = entity.id();
+    final var creationTime = entity.creationTime();
+    final var lastAccessedTime = entity.lastAccessedTime();
+    final var maxInactiveIntervalInSeconds = entity.maxInactiveIntervalInSeconds();
+    final var attributes = deserializeSessionAttributes(entity);
+
+    final var webSession = new WebSession(sessionId);
+    webSession.setCreationTime(toInstant(creationTime));
+    webSession.setLastAccessedTime(toInstant(lastAccessedTime));
+    webSession.setMaxInactiveInterval(toDuration(maxInactiveIntervalInSeconds));
```

Comment:
In my opinion, it can't be null; Spring will set it to the proper value (which can be configured, I guess). As mentioned above, I will remove the null-check.